File: Python3/3111_minimum_rectangles_to_cover_points.py
# https://leetcode.cn/problems/minimum-rectangles-to-cover-points/

class Solution:
    def minRectanglesToCoverPoints(self, points, w):
        """
        Inputs:
            points: list[list[int]]
            w: int
        Outputs:
            res: int
        """
        points.sort()
        res = 1
        end = points[0][0] + w
        for x, y in points:
            if x > end:
                res += 1
                end = x + w
        return res

File: Cpp/3111_minimum_rectangles_to_cover_points.cpp
// https://leetcode.cn/problems/minimum-rectangles-to-cover-points/
        
class Solution {
public:
    int minRectanglesToCoverPoints(vector<vector<int>>& points, int w) {
        sort(points.begin(), points.end());
        int res = 1;
        int end = points[0][0] + w;
        for (auto &p: points) {
            if (p[0] > end) {
                res += 1;
                end = p[0] + w;
            }
        }
        return res;
    }
};
